subsys/testsuite/ztest/include/zephyr/ztest_assert.h

```diff
@@ -282,6 +282,13 @@ static inline bool z_zexpect(bool cond, const char *default_msg, const char *fil
  */
 #define zassert_ok(cond, ...) zassert(!(cond), #cond " is non-zero", ##__VA_ARGS__)
 
+/**
+ * @brief Assert that @a cond is not 0 (failure)
+ * @param cond Condition to check
+ * @param ... Optional message and variables to print if the assertion fails
+ */
+#define zassert_nok(cond, ...) zassert(!!(cond), #cond " is zero", ##__VA_ARGS__)
```

Comment:
As a personal **non**-blocking preference, I'd rather use `!` instead of adding another set of macros to maintain.

Reply:
So you mean something like `zassert_ok(!ret);`? I personally prefer the solution with `nok`, resp. `not_ok`, because the `!` can also be missed.